### PR TITLE
Configuration Redux

### PR DESCRIPTION
**Pull request by eldani17**
## Summary

- Libraries are added: redux, react-redux.
- The provider that wraps the entire app is added.
- The folder and file structure for Redux is created.
- A test of the store is made with reactotron-redux.

## Screenshots
Screenshot from 2021-03-22 16-17-44

## Trello Card

https://trello.com/c/QwGS33Je/25-n-91-redux-i-actions-reducers-store

### CODE REVIEW

File: wbooks/src/redux/auth/actions.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  user: string;
+  password: string;
+}
+
+export const loginAuth = ({ user, password }: login) => {
+  return {
+    type: TypeAuth.LOGIN,
+    payload: { user, password }
+  };
+};
```
Comment (mattgle) [inline wbooks/src/redux/auth/actions.ts]:
```suggestion
export const loginAuth = ({ user, password }: login) => ({
    type: TypeAuth.LOGIN,
    payload: { user, password }
  });
```

File: wbooks/src/redux/auth/reducer.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { Action } from '@interfaces/action';
+
+import { TypeAuth } from './actions';
+
+const initialState = () => {
+  return { login: false };
+};
```
Comment (mattgle) [inline wbooks/src/redux/auth/reducer.ts]:
```suggestion
const initialState = () => ({
   login: false 
});
```

File: wbooks/src/redux/book/actions.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    payload: book
+  };
+};
+
+export const bookList = (list: Book[]) => {
+  return {
+    type: TypeBook.LIST,
+    payload: list
+  };
+};
```
Comment (mattgle) [inline wbooks/src/redux/book/actions.ts]: same comment here.

File: wbooks/src/app/screens/bookList/index.tsx [hunk 4]
```diff
@@ -3,9 +3,16 @@ import { FlatList, ListRenderItem, SafeAreaView, View } from 'react-native';
...
+import { connect } from 'react-redux';
 
 import styles from './styles';
 
+const mapStateToProps = (state: any) => {
+  return {
+    book: state.book
+  };
+};
+
```
Comment (mattgle) [inline wbooks/src/app/screens/bookList/index.tsx]: same comment

File: wbooks/src/app/screens/bookList/index.tsx [hunk 5]
```diff
@@ -3,9 +3,16 @@ import { FlatList, ListRenderItem, SafeAreaView, View } from 'react-native';
 import { BOOKS_MOCK } from '@constants/mockBooks';
 import Book from '@components/book';
 import { Book as IBook } from '@interfaces/bookInterface';
+import { connect } from 'react-redux';
 
 import styles from './styles';
 
+const mapStateToProps = (state: any) => {
```
Comment (mattgle) [inline wbooks/src/app/screens/bookList/index.tsx]: you can create an interface of State, create the file reduxInterfaces.ts

**Review by mattgle — CHANGES_REQUESTED** [review]
Just a few changes, nice work!

File: wbooks/src/redux/auth/actions.ts (hunk 1 above)
Comment (eldani17) [reply wbooks/src/redux/auth/actions.ts]: Modified return.

File: wbooks/src/redux/auth/reducer.ts (hunk 2 above)
Comment (eldani17) [reply wbooks/src/redux/auth/reducer.ts]: Modified return.

File: wbooks/src/redux/book/actions.ts (hunk 3 above)
Comment (eldani17) [reply wbooks/src/redux/book/actions.ts]: Modified return.

File: wbooks/src/app/screens/bookList/index.tsx (hunk 4 above)
Comment (eldani17) [reply wbooks/src/app/screens/bookList/index.tsx]: Modified return.

File: wbooks/src/app/screens/bookList/index.tsx (hunk 5 above)
Comment (eldani17) [reply wbooks/src/app/screens/bookList/index.tsx]: Any was changed to Interface

**Review by mattgle — APPROVED** [review]

**Pull request merged**